File: linear_search.c
// Linear Search in C

#include <stdio.h>

int search(int array[], int n, int x) {
  
  // Going through array sequencially
  for (int i = 0; i < n; i++)
    if (array[i] == x)
      return i;
  return -1;
}

int main() {
  int array[] = {2, 4, 0, 1, 9};
  int x = 1;
  int n = sizeof(array) / sizeof(array[0]);

  int result = search(array, n, x);

  (result == -1) ? printf("Element not found") : printf("Element found at index: %d", result);
}

File: Linear_search.js
// Javascript code to linearly search x in arr[]. 

  

function search(arr, n, x) 
{ 

    for (let i = 0; i < n; i++) 

        if (arr[i] == x) 

            return i; 

    return -1; 
} 

  
// Driver code 

  

    let arr = [ 2, 3, 4, 10, 40 ]; 

    let x = 10; 

    let n = arr.length; 

  

    // Function call 

    let result = search(arr, n, x); 

    (result == -1) 

        ? console.log("Element is not present in array") 

        : console.log("Element is present at index " + result); 
